### PR TITLE
MAINT: compat with new pandas and numpydoc; fix deprecation warnings

### LINKED ISSUE #1497: bug: make test fails in travis ci (python 3.6?)
Dear all,

I found that my PR #1495 fail to pass the `travis CI` test for a problem related to the conda environment. Briefly, by inspecting the travis output I noticed that modules were tested on a python `3.6` environment, even if `3.5` version is specified in configuration. The commands to repruduce error locally are the following:
```bash
$ conda create -n travis python=3.6
$ source activate travis
$ conda install --file ci/conda_requirements.txt
$ pip install -r ci/pip_requirements.txt
$ pip install . --no-deps
$ WITH_COVERAGE=TRUE make test
```
This will raise the same errors raised by `Travis CI`. By inspecting code, I found that the problem arises from `skbio.stats.power`, at line [1076](https://github.com/biocore/scikit-bio/blob/master/skbio/stats/power.py#L1076): The order of items of `collections.Counter(set_pos)` is different from python `3.5` to `3.6`. Here is a sample code to reproduce the error:
```python
import collections
set_pos = [2,0,3,1,3,2,2,0,4,3,2,1,5,0,0,3,0,3,3,1,1]
collections.Counter(set_pos).items()
# in python 3.5
# dict_items([(0, 5), (1, 4), (2, 4), (3, 6), (4, 1), (5, 1)])
# in python 3.6
# dict_items([(2, 4), (0, 5), (3, 6), (1, 4), (4, 1), (5, 1)])
```
I think i've fixed the issue in python code, I will open a new PR. Another solution could be fixing `.travis.yml` to force python `3.5` installation, however make test will fail in `3.6` enviroment.
Thank you for your support,
regards,
Paolo

### LINKED ISSUE #1511: Build errors with Sphinx and Pandas
We're observing build errors with Sphinx. This is causing PRs to indicate failures in CI which are unrelated to the PR itself. #1510 is an exemplar of the problem.

On a local build, using Sphinx 1.5.5, I am able to build the docs.

### LINKED ISSUE #1531: Test failure on Debian system
Hi,
I'm one of the Debian maintainers of scikit-bio.  There was a [failure in the test suite](https://bugs.debian.org/868962) reported.  Do you have any hint how to fix this?
Kind regards, Andreas.

### PR DESCRIPTION
**Pull request by jairideout**
Made the following compatibility updates to fix the longstanding Travis-CI failures and make the codebase a little more futureproof:

- The codebase is now compatible with pandas >=0.19.2, including the latest pandas release (0.20.3), which introduced a number of backwards-incompatible changes.

- Fixed doc build failures by unpinning numpydoc version (latest PyPI numpydoc release is now compatible with latest Sphinx release).

- Fixed all deprecation warnings coming from third-party packages -- most were pandas deprecations, and one was from `scipy.stats`.

- Fixed docs in a couple of places that weren't compatible with numpydoc.

Fixes #1497
Fixes #1511
Fixes #1531

---

Please complete the following checklist:

* [x] I have read the guidelines in [CONTRIBUTING.md](https://github.com/biocore/scikit-bio/blob/master/CONTRIBUTING.md).

* [x] I have documented all public-facing changes in [CHANGELOG.md](https://github.com/biocore/scikit-bio/blob/master/CHANGELOG.md).

* [ ] **This pull request includes code, documentation, or other content derived from external source(s).** If this is the case, ensure the external source's license is compatible with scikit-bio's [license](https://github.com/biocore/scikit-bio/blob/master/COPYING.txt). Include the license in the `licenses` directory and add a comment in the code giving proper attribution. Ensure any other requirements set forth by the license and/or author are satisfied. **It is your responsibility to disclose code, documentation, or other content derived from external source(s).** If you have questions about whether something can be included in the project or how to give proper attribution, include those questions in your pull request and a reviewer will assist you.

* [x] **This pull request does not include code, documentation, or other content derived from external source(s).**

**Note:** [REVIEWING.md](https://github.com/biocore/scikit-bio/blob/master/REVIEWING.md) may also be helpful to see some of the things code reviewers will be verifying when reviewing your pull request.

### CODE REVIEW

**Comment by jairideout** [comment]
Looking into the doc build failures, forgot to try building them locally.

**Comment by ebolyen** [comment]
Can I just say I'm really sad that there isn't a commit that says: "Nixed the .ix"?

**Comment by ElDeveloper** [comment]
Test pass 🎉 ✨

**Comment by jairideout** [comment]
Ready for review! Knocked out a couple more issues, see PR description (or the amended commit) for updates of what's fixed.

File: skbio/stats/gradient.py [hunk 1]
```diff
@@ -428,7 +430,7 @@ def __init__(self, coords, prop_expl, metadata_map,
                              % (len(prop_expl), axes))
 
         # Restrict coordinates to those axes that we actually need to compute
-        self._coords = coords.ix[:, :axes-1]
+        self._coords = coords.loc[:, :axes-1]
```
Comment (ebolyen) [inline skbio/stats/gradient.py]: This should be `iloc`

File: ci/pip_requirements.txt [hunk 2]
```diff
@@ -5,5 +5,5 @@ lockfile
 CacheControl
 git+git://github.com/sphinx-doc/sphinx.git
 sphinx-bootstrap-theme
-git+git://github.com/numpy/numpydoc.git@1a848331c2cf53d4fe356f4607799524bcc577ed
```
Comment (ebolyen) [inline ci/pip_requirements.txt]: 🎉

File: skbio/metadata/_mixin.py [hunk 3]
```diff
@@ -368,7 +371,15 @@ def __deepcopy__(self, memo):
...
         if self.has_positional_metadata():
-            return copy.deepcopy(self.positional_metadata, memo)
+            # `copy.deepcopy` no longer recursively copies contents of the
+            # DataFrame, so we must handle the deep copy ourselves.
+            # Reference: https://github.com/pandas-dev/pandas/issues/17406
+            df = self.positional_metadata
+            data_cp = copy.deepcopy(df.values.tolist(), memo)
+            return pd.DataFrame(data_cp,
+                                index=df.index.copy(deep=True),
+                                columns=df.columns.copy(deep=True),
```
Comment (ebolyen) [inline skbio/metadata/_mixin.py]: These are made of hashable objects right? So even a shallow copy is deep enough for our purposes?

File: skbio/stats/gradient.py (hunk 1 above)
Comment (jairideout) [reply skbio/stats/gradient.py]: It doesn't work with `.iloc` because the column names are integers.

File: skbio/metadata/_mixin.py (hunk 3 above)
Comment (jairideout) [reply skbio/metadata/_mixin.py]: They're documented to only contain hashable objects but in practice you can create `Index` objects with non-hashables, so this seemed safer to do.

**Comment by jairideout** [comment]
@ebolyen thanks for reviewing! I responded to your comments.

File: skbio/stats/gradient.py (hunk 1 above)
Comment (jairideout) [reply skbio/stats/gradient.py]: A little more detail: for whatever reason, using `.loc` here with integer column names matches the behavior of `.ix` so I decided to go with that.

**Comment by ebolyen** [comment]
Thanks @jairideout!!!

**Comment by wasade** [comment]
Awesome, thank you so much @jairideout!!!

**Comment by jairideout** [comment]
No problem guys! :)

**Pull request merged**